libmultilabel/metrics.py

```diff
@@ -1,9 +1,72 @@
 import re
 
+import torch
 import numpy as np
 from torchmetrics import Metric, MetricCollection, F1, Precision, Recall
 
 
+class RPrecision(Metric):
+    def __init__(
+        self,
+        top_k,
+        dist_sync_on_step=False
+    ):
+        super().__init__(dist_sync_on_step=dist_sync_on_step)
+        self.top_k = top_k
+        self.add_state("score", default=torch.tensor(0.), dist_reduce_fx="sum")
+        self.add_state("num_sample", default=torch.tensor(0), dist_reduce_fx="sum")
+
+    def update(self, preds, target):
+        assert preds.shape == target.shape
+        for y_pred, y_true in zip(preds, target):
+            self.score += self.ranking_rprecision_score(y_pred, y_true)
+        self.num_sample += len(preds)
+
+    def compute(self):
+        return self.score.float() / self.num_sample
+
+    def ranking_rprecision_score(self, y_pred, y_true):
+        n_pos = torch.sum(y_true == 1)
+        order = torch.argsort(-y_pred)
```

Comment:
torchmetrics has something called select_topk??
Also I don't think this function is needed. At line 22 we should avoid a loop and handle the batch together??

Reply:
Yes. I followed lmtc's. Now there's no for loop in `RPrecision`.